src/org/jitsi/impl/neomedia/codec/video/h264/DePacketizer.java

```diff
@@ -519,6 +537,105 @@ else if (nal_unit_type == 28) // FU-A Fragmentation unit (FU)
     }
 
     /**
+     * Returns true if the buffer contains a H264 key frame at offset
+     * <tt>offset</tt>.
+     *
+     * @param buff the byte buffer to check
+     * @param off the offset in the byte buffer where the actuall data starts
+     * @param len the length of the data in the byte buffer
+     * @return true if the buffer contains a H264 key frame at offset
+     * <tt>offset</tt>.
+     */
+    public boolean isKeyFrame(byte[] buff, int off, int len)
+    {
+      if (buff == null || len < 1 || buff.length < len)
+      {
+          return false;
+      }
+
+      int nalType =  buff[off] & kTypeMask;
+      // Single NAL Unit Packet
+      if (nalType == kFuA)
+      {
+          // Fragmented NAL units (FU-A).
+          if (parseFuaNaluForKeyFrame(buff, off, len))
+          {
+              return true;
+          }
+      }
+      else
+      {
+          if (parseSingleNaluForKeyFrame(buff, off, len))
+          {
+              return true;
+          }
+      }
+
+      return false;
+    }
+
+    /**
+     * Checks if a a fragment of a NAL unit from a specific FU-A RTP packet
+     * payload is keyframe or not
+     */
+    private Boolean parseFuaNaluForKeyFrame(byte[] buff, int off, int len) {
+      if (len < kFuAHeaderSize)
+      {
+          return false;
+      }
+      return ((buff[off + 1] & kTypeMask) == kIdr);
+    }
+
+    /**
+     * Checks if a a fragment of a NAL unit from a specific FU-A RTP packet
+     * payload is keyframe or not
+     */
+    private boolean parseSingleNaluForKeyFrame(byte[] buff, int off, int len)
+    {
+        int naluStart = off + kNalHeaderSize;
+        int naluLength = len - kNalHeaderSize;
+        int nalType = buff[off] & kTypeMask;
+        if (nalType == kStapA)
+        {
+            // Skip the StapA header (StapA nal type + length).
+            if (len <= kStapAHeaderSize)
+            {
+                logger.error("StapA header truncated.");
+                return false;
+            }
+            if (!verifyStapANaluLengths(buff, naluStart, naluLength)) {
+                logger.error("StapA packet with incorrect NALU packet lengths.");
+                return false;
+            }
+            nalType = buff[off + kStapAHeaderSize] & kTypeMask;
+        }
+        return (nalType == kIdr || nalType == kSps || nalType == kPps || nalType == kSei);
+    }
+
+    private static boolean verifyStapANaluLengths(byte[] data, int offset, int lengthRemaining)
```

Comment:
This line (and a few others) exceed our 80 chars limit. If you're using IntelliJ Idea you can configure it to show a gray column at the 80 chars limit (I believe the option is called right margin). You can do the same thing with ViM (and most likely any other editor)

    set colorcolumn=80,100